notify/notify.go

```diff
@@ -442,14 +434,18 @@ func allAlertsResolved(alerts []*types.Alert) bool {
 	return true
 }
 
-func (n *DedupStage) needsUpdate(entry *nflogpb.Entry, hash []byte, resolved bool, repeat time.Duration) (bool, error) {
+func (n *DedupStage) needsUpdate(entry *nflogpb.Entry, firing, resolved map[uint64]struct{}, repeat time.Duration) (bool, error) {
 	// If we haven't notified about the alert group before, notify right away
 	// unless we only have resolved alerts.
 	if entry == nil {
-		return !resolved, nil
+		return ((len(firing) > 0) || (n.sendResolved && len(resolved) > 0)), nil
```

Comment:
Wondering here. I remember providing explicit logic about not sending resolved notifications about stuff we have never notified about being firing in the first place.

But maybe just keep as it is here – might be better in doubt in case some other AM notified something as firing but couldn't propagate the entry via the mesh and then died.